ecl/hqlcpp/hqlres.cpp

```diff
@@ -613,22 +613,28 @@ bool ResourceManager::flush(StringBuffer &filename, const char *basename, bool f
         unsigned id = s.id;
         VStringBuffer binfile("%s_%s_%u.bin", filename.str(), type, id);
         VStringBuffer label("%s_%u_txt_start", type, id);
+
+        //Note on targets where the @ character is the start of a comment (eg ARM) then another character is used instead. For example the ARM port uses the % character.
+#if defined(__arm__) || defined(__aarch64__)
+        const char typePrefix = '%';
+#else
+        const char typePrefix = '@';
+#endif
+
         if (compilerType == ClangCppCompiler)
         {
 #ifdef __APPLE__
             if (id <= 1200)  // There is a limit of 255 sections before linker complains - and some are used elsewhere
```

Comment:
I know not new but I don't understand this line or the comment. Why 1200 not 256 and why is it ok to just omit the line once we get to 1200?